{{cookiecutter.project_slug}}/setup_env.sh

```diff
@@ -1,61 +1,109 @@
 #!/bin/bash
+#
+# Create conda environment with pinned or unpinned run and/or dev requirements
+#
+# - 2022-08 (D. Regenass) Write original script
+# - 2022-09 (S. Ruedisuehli) Refactor; add some options
+#
 
-### OPTION DEFAULTS ###
-ENV_NAME={{cookiecutter.project_slug}}
+# Default options
+ENV_NAME="{{ cookiecutter.project_slug.replace("_", "-") }}"
 PYVERSION=3.10
+PINNED=true
 DEV=false
-PINNED=false
-#GET OPTIONS FROM COMMAND LINE ARGS
-while getopts n:v:dph flag
-do
+EXPORT=false
+INSTALL=false
+FORCE=false
+CONDA=conda
+HELP=false
+
+help="Usage: $(basename "${0}") [-n NAME] [-P VER] [-u] [-e] [-d] [-i] [-f] [-m] [-h]
+
+Options:
+ -n NAME    Env name (-d adds -dev) [default: ${ENV_NAME}]
+ -P VER     Python version [default: ${PYVERSION}]
+ -u         Use unpinned requirements (minimal version restrictions)
+ -e         Export environment files (requires -u)
+ -d         Install additional dev requirements
+ -i         Install package itself (editable with -d)
+ -f         Force overwrite of existing env
+ -m         Use mamba instead of conda
+ -h         Print this help message and exit
+"
+
+# Eval command line options
+while getopts n:P:defhimu flag; do
     case ${flag} in
         n) ENV_NAME=${OPTARG};;
-        v) PYVERSION=${OPTARG};;
+        P) PYVERSION=${OPTARG};;
```

Comment:
Initially, `-p` was already used for 'pinned', but this would no longer be the case if we make that the default and use `-u` for unpinned instead. In that case, `-p` would definitely be better.``

Reply:
Done.